### PR TITLE
Addresses #4615, wrap OutputControl:Table:Style and Output:SQLite

### LINKED ISSUE #4615: Wrap OutputControl:Table:Style and Output:SQLite
Enhancement Request
--------------
https://github.com/NREL/EnergyPlus/pull/8474 allowed for different units between the energyplus tabular output and the output SQL file. Since the openstudio_results measure requires the SQL file in SI units, this allows for generating both the tabular output and the openstudio standard output in IP units with one run (see [this](https://unmethours.com/question/76065/output-units-of-eplustbl-and-openstudio_results/) UnmetHours question). 

It's easy enough to make the required input changes with an EnergyPlus measures, but would be _even easier_ if those objects were wrapped. Or maybe set thru a FT option?

### PR DESCRIPTION
**Pull request by joseph-robertson**
Pull request overview
---------------------

 - Fixes #4615
 - Also adds FT tests (missing?) and RT support for `OutputControlReportingTolerances`

### Pull Request Author

 - [x] Model API Changes / Additions
 - [x] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [x] Model API methods are tested (in `src/model/test`)
 - [x] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [x] If a new object or method, added a test in NREL/OpenStudio-resources: https://github.com/NREL/OpenStudio-resources/pull/171
 - [x] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [x] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [x] All new and existing tests passes
 - [x] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [x] If change to an IDD file, add the label `IDDChange`
 - [x] If breaking existing API, add the label `APIChange`
 - [x] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] Code Style, strip trailing whitespace, etc.
 - [x] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [x] Labeling is ok
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [x] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified

### CODE REVIEW

File: src/energyplus/ForwardTranslator.cpp [hunk 1]
```diff
@@ -3278,9 +3292,11 @@ namespace energyplus {
     result.push_back(IddObjectType::OS_ZoneCapacitanceMultiplier_ResearchSpecial);
     result.push_back(IddObjectType::OS_OutputControl_Files);
     result.push_back(IddObjectType::OS_OutputControl_ReportingTolerances);
+    result.push_back(IddObjectType::OS_OutputControl_Table_Style);
     result.push_back(IddObjectType::OS_Output_DebuggingData);
     result.push_back(IddObjectType::OS_Output_Diagnostics);
     result.push_back(IddObjectType::OS_Output_JSON);
+    result.push_back(IddObjectType::OS_Output_SQLite);
```
Comment (jmarrec) [inline src/energyplus/ForwardTranslator.cpp:3299]: Add them to the list of objects to always translate, makes sense.

File: src/energyplus/ForwardTranslator.cpp [hunk 2]
```diff
@@ -4361,13 +4377,16 @@ namespace energyplus {
...
 
-  void ForwardTranslator::createStandardOutputRequests() {
+  void ForwardTranslator::createStandardOutputRequests(const model::Model& model) {
     if (!m_excludeHTMLOutputReport) {
-      IdfObject tableStyle(IddObjectType::OutputControl_Table_Style);
-      m_idfObjects.push_back(tableStyle);
-      tableStyle.setString(OutputControl_Table_StyleFields::ColumnSeparator, "HTML");
-      if (m_ipTabularOutput) {
-        tableStyle.setString(OutputControl_Table_StyleFields::UnitConversion, "InchPound");
+      boost::optional<model::OutputControlTableStyle> _tableStyle = model.getOptionalUniqueModelObject<model::OutputControlTableStyle>();
```
Comment (jmarrec) [inline src/energyplus/ForwardTranslator.cpp:4382]: (No changes needed, but FYI if we end up making our IDD (or Ctor) use the same defaults as the FT, you could just call model.getUniqueModelObject<XXX>() to instantiate the object as we operate on a copy of our model to begin with. This is maybe clearer this way though.)

File: src/energyplus/Test/OutputControlReportingTolerances_GTest.cpp [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+using namespace openstudio;
+
+TEST_F(EnergyPlusFixture, ForwardTranslator_OutputControlReportingTolerances) {
+
+  ForwardTranslator ft;
+
+  // Create a model
+  Model m;
+
+  // Get the unique object
```
Comment (jmarrec) [inline src/energyplus/Test/OutputControlReportingTolerances_GTest.cpp:59]: We probably want to first do a translation without the object instantiated, ensure one isn't present in the IDF.

File: src/energyplus/Test/OutputControlTableStyle_GTest.cpp [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+TEST_F(EnergyPlusFixture, ForwardTranslator_OutputControlTableStyle) {
+
+  ForwardTranslator ft;
+
+  // Create a model
+  Model m;
+
+  // Get the unique object
+  OutputControlTableStyle outputControlTableStyle = m.getUniqueModelObject<OutputControlTableStyle>();
```
Comment (jmarrec) [inline src/energyplus/Test/OutputControlTableStyle_GTest.cpp:60]: , make sure we do get the default behavior, also test with setExcludeHTMLOutputReport / setIPTabularOutput as well.

File: src/energyplus/Test/OutputSQLite_GTest.cpp [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+TEST_F(EnergyPlusFixture, ForwardTranslator_OutputSQLite) {
+
+  ForwardTranslator ft;
+
+  // Create a model
+  Model m;
+
+  // Get the unique object
+  OutputSQLite outputSQLite = m.getUniqueModelObject<OutputSQLite>();
```
Comment (jmarrec) [inline src/energyplus/Test/OutputSQLite_GTest.cpp:60]: Test without object instantiated, make sure we do get the default behavior, and test with `ft.setExcludeSQliteOutputReport(bool)` too

**Review by jmarrec — CHANGES_REQUESTED** [review]
This is looking really good. I added a couple of minor suggestions.

File: src/energyplus/ForwardTranslator/ForwardTranslateOutputControlTableStyle.cpp [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+  boost::optional<IdfObject> ForwardTranslator::translateOutputControlTableStyle(OutputControlTableStyle& modelObject) {
+    IdfObject idfObject(openstudio::IddObjectType::OutputControl_Table_Style);
+
+    m_idfObjects.push_back(idfObject);
+
+    OptionalString s = modelObject.name();
+    if (s) {
+      idfObject.setName(*s);
+    }
```
[the remark this answers is not in the document]
Comment (joseph-robertson) [reply src/energyplus/ForwardTranslator/ForwardTranslateOutputControlTableStyle.cpp:54]: Neither does `OutputControlReportingTolerances`... I will remove there, too.

File: src/model/OutputSQLite.cpp [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+  bool OutputSQLite::setUnitConversionforTabularData(const std::string& unitConversionforTabularData) {
+    return getImpl<detail::OutputSQLite_Impl>()->setUnitConversionforTabularData(unitConversionforTabularData);
+  }
+
+  /// @cond
+  OutputSQLite::OutputSQLite(std::shared_ptr<detail::OutputSQLite_Impl> impl) : ModelObject(impl) {}
+  OutputSQLite::OutputSQLite(Model& model) : ModelObject(OutputSQLite::iddObjectType(), model) {
+    setOptionType("SimpleAndTabular");
+    setUnitConversionforTabularData("UseOutputControlTableStyle");
+  }
```
[the remark this answers is not in the document]
Comment (joseph-robertson) [reply src/model/OutputSQLite.cpp:128]: Hmm it's like this for a bunch of `OutputXXX` objects...

[the remark this answers is not in the document]
Comment (jmarrec) [reply src/model/OutputSQLite.cpp:128]: Yeah I know. Generateclass.rb does it I think. It's nitpicking anyways

File: src/energyplus/Test/OutputControlReportingTolerances_GTest.cpp [hunk 8]
```diff
@@ -51,26 +51,36 @@ using namespace openstudio;
...
+    WorkspaceObjectVector idfObjs = w.getObjectsByType(IddObjectType::OutputControl_ReportingTolerances);
+    ASSERT_EQ(0u, idfObjs.size());
+  }
 
-  // Get the unique object
-  OutputControlReportingTolerances outputControlReportingTolerances = m.getUniqueModelObject<OutputControlReportingTolerances>();
+  // It's there
+  {
+    Model m;
+    ForwardTranslator ft;
```
Comment (jmarrec) [inline src/energyplus/Test/OutputControlReportingTolerances_GTest.cpp:67]: (could use the same ft and model).

**Review by jmarrec — APPROVED** [review]

**Pull request merged**